### PR TITLE
Introduce OTREntityTranscoder and GenericMessageRequestStrategy

### PR DESCRIPTION
**Pull request by typfel**
Adds `OTREntityTranscoder` as a base transcoder for client handling and
`GenericMessageRequestStrategy` for sending any generic message in a
conversation.

depends on:
- [x] https://github.com/wireapp/wire-ios-data-model/pull/93
- [x] https://github.com/wireapp/wire-ios-request-strategy/pull/10

### CODE REVIEW

File: Source/Helpers/OTREntity.swift [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+public protocol OTREntity : DependencyEntity {
+    
+    var conversation : ZMConversation { get }
+    
+}
+
+extension OTREntity {
+    
+    public var dependentObjectNeedingUpdateBeforeProcessingOTREntity: AnyObject? {
+        // FIXME this should be shared with OTRMessage once it also implements the OTREntity protocol
```
Comment (daehn) [inline Source/Helpers/OTREntity.swift]: Should this be done in this PR or will there be another on resolving this `FIXME`?

File: Source/Transcoders/OTREntityTranscoder.swift [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+/// HTTP status of a request that has
```
Comment (daehn) [inline Source/Transcoders/OTREntityTranscoder.swift]: The comment is missing some information

File: Source/Helpers/OTREntity.swift (hunk 1 above)
Comment (typfel) [inline Source/Helpers/OTREntity.swift]: This will be done in another PR since it requires re-factoring of modified / inserted - sync, which is quite a big task.

**Comment by daehn** [comment]
💣 Conflicts
